.github/workflows/pull-request-checks.yaml

```diff
@@ -391,7 +391,7 @@ jobs:
           # This is needed in addition to -yq to prevent apt-get from asking for
           # user input
           DEBIAN_FRONTEND: noninteractive
-        run: sudo apt-get install --no-install-recommends -y g++ binutils flex bison cmake maven jq libxml2-utils openjdk-11-jdk-headless lcov
+        run: sudo apt-get install -y g++ gcc binutils flex bison cmake maven jq libxml2-utils openjdk-11-jdk-headless lcov
```

Comment:
I think you're doing yourself a disservice in removing `--no-install-recommends`. Perhaps use `build-essential` to get all of `g++`, `gcc`, `binutils` and any other packages really required for building.

Reply:
Hi Michael, why do you think it's a disservice to ourselves to remove `--no-install-recommends`? My rationale for removing it is that by keeping it here we may be subtly missing dependencies that are recommended by the system, but they really are essential for our build process, and by missing them we have all sorts of errors.

I guess this makes the process slightly more opaque (in terms of what dependencies will be installed, because they can be changed under our feet) but it's making it safer in terms of making the build much less likely to fail because of missed dependencies. What do you think?

Reply:
I'm not entirely sure I understand whether you argue for or against `--no-install-recommends`, but let me try to clarify my position:
1) We should (ourselves) have a very clear understanding of the dependencies so that we can correctly document these in files like `COMPILING.md`. We cannot know whether our customers use/don't use `--no-install-recommends` (or their equivalents in APT configuration files). That said, a GitHub action need not be the one place to test our own understanding of dependencies, but it might well be a good one.
2) Installing recommended packages might make it easier to get the package lists right, but will likely also install all sorts of packages that we do not need. Installing packages takes time, and CI jobs shouldn't be wasting time.

Reply:
Hm, I see your point. I'm convinced by point 1., so I'll re-introduce the option as soon as I stabilise an issue I'm having with CI.

Re 2., I can also see this point, but any time savings are marginal. As things are, it downloads about 300MB of packages in about 7 seconds, and it takes another 10-15 seconds to unpack those. At the same time, we take more than 30 minutes to compile and test.

Reply:
Though the more separate things there are to download, the greater the chance of random network gliches killing a build. GitHub actions seems a lot more reliable than, say, Travis was when we used that, but still worth minimising your 'surface area' for glitches :-)